roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -401,6 +406,66 @@ static void deviceTwinCallback(DEVICE_TWIN_UPDATE_STATE update_state, const unsi
     json_value_free(root_value);
 }
 
+static bool ReadKeyFromFile(const std::string fileName, std::string &key)
+{
+    //key = NULL;
+    if (fileName.empty())
+    {
+        ROS_ERROR("File name empty.");
+        return false;
+    }
+
+    std::ifstream f(fileName); //taking file as inputstream
+    if(f) {
+        std::ostringstream ss;
+        ss << f.rdbuf(); // reading data
+        key = ss.str();
+    }
+    else
+    {
+        ROS_ERROR("Could not read from file.");
+        return false;
+    }
+
+    ROS_INFO("File Read: %s", fileName.c_str());
+    return true;
+}
+
+static bool InitializeX509Certificate(IOTHUB_DEVICE_CLIENT_HANDLE deviceHandle, ros::NodeHandle nh)
+{
+    std::string x509certificateFile;
+    std::string x509certificate;
+    nh.getParam("public_key_file", x509certificateFile);
+    ROS_INFO("public_key_file: %s", x509certificateFile.c_str());
+    if (!ReadKeyFromFile(x509certificateFile, x509certificate))
+    {
+        ROS_ERROR("Could not read x509 certificate/public key file, aborting.");
+        return false;
+    }
+
+    std::string x509privatekeyFile;
+    std::string x509privatekey;
+    nh.getParam("private_key_file", x509privatekeyFile);
+    if (!ReadKeyFromFile(x509privatekeyFile, x509privatekey))
+    {
+        ROS_ERROR("Could not read x509 private key file, aborting.");
+        return false;
+    }
+
+    // Set the X509 certificates in the SDK
+    if (
+        (IoTHubDeviceClient_SetOption(deviceHandle, OPTION_X509_CERT, x509certificate.c_str()) != IOTHUB_CLIENT_OK) ||
+        (IoTHubDeviceClient_SetOption(deviceHandle, OPTION_X509_PRIVATE_KEY, x509privatekey.c_str()) != IOTHUB_CLIENT_OK)
+        )
+    {
+        ROS_ERROR("failure to set options for x509, aborting\r\n");
```

Comment:
nit: remove `\r\n`.

Reply:
Would also be good to output the error from SetOption